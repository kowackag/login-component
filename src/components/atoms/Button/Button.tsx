export const Button = ({ text, type }: { text: string; type: HTMLButtonElement["type"] }) => {
	return (
		<button
			className="my-2 w-full rounded-lg bg-bright-rose text-center font-primaryMedium text-sm uppercase leading-10 tracking-widest text-neutral-100 shadow-btn"
			type={type}
		>
			{text}
		</button>
	);
};
