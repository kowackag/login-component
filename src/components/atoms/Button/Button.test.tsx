import { expect, it, describe } from "vitest";
import { render, screen } from "@testing-library/react";
import { Button } from "./Button";

describe("Button", () => {
	it("Button displays appropriate text", () => {
		render(<Button text="Login" type="submit" />);

		const button = screen.getByRole("button");
		expect(button).toHaveTextContent("Login");
	});

	it("Button has appropriate type", () => {
		render(<Button text="Login" type="submit" />);

		const button = screen.getByRole("button");
		expect(button).toHaveAttribute("type", "submit");
	});

	it("Button renders correctly", () => {
		const wrapper = render(<Button text="Login" type="submit" />)
		expect(wrapper.container).toMatchSnapshot()
	})
});
