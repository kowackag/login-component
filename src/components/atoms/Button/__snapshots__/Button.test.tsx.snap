// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > Button renders correctly 1`] = `
<div>
  <button
    class="my-2 w-full rounded-lg bg-bright-rose text-center font-primaryMedium text-sm uppercase leading-10 tracking-widest text-neutral-100 shadow-btn"
    type="submit"
  >
    Login
  </button>
</div>
`;
